docs/components/table.md

```diff
@@ -85,6 +85,80 @@ Lessen the table cell padding
   isSmall />
 ```
 
+### Sorting
+There are two props used to make the table sortable; ```sortState```, which is either 'ascending' or 'descending' and ```sortKey```, which tells the table which column is currently being sorted. If a sortKey exists, then clicking the
+table header will emit an Event called ```sortField``` which must be handled by the parent component to implement the
+actual sorting logic. An arrow then appears beside the table header, the state of the arrow depending on the sortState.
+In the following example, there's some basic logic in the parent component that implements the sort function, and the table is able to be sorted by the name by clicking the name header.
+
+<template>
+  <KTable
+    :options="tableOptions"
+    :sort-state="sortState"
+    sort-key="name"
+    @sort-field="sortField"
+    />
+</template>
+
+```vue
+<template>
+<KTable
+    :options="tableOptions"
+    :sort-state="sortState"
+    sort-key="name"
+    @sort-field="sortField"
+    />
+</template>
+<script>
+export default {
+  data() {
+    return {
+      sortState: 'ascending',
+      tableOptions: {
+        headers: [
+          { label: 'Name', key: 'name' },
+          { label: 'ID', key: 'id' },
+          { label: 'Enabled', key: 'enabled' },
+          { key: 'actions', hideLabel: true }
+        ],
+        data: [
+          {
+            name: 'Basic Auth',
+            id: '517526354743085',
+            enabled: 'true'
+          },
+          {
+            name: 'Website Desktop',
+            id: '328027447731198',
+            enabled: 'false'
+          },
+          {
+            name: 'Android App',
+            id: '405383051040955',
+            enabled: 'true'
+          }
+        ]
+      }
+    }
+  },
+  methods: {
+    sortField (key) {
+      if (this.sortState === 'ascending') {
+        this.tableOptions.data.sort((a, b) => {
+          if (a[key] < b[key]) return -1
+          if (a[key] > b[key]) return 1
+          return 0
+        })
+      } else {
+        this.tableOptions.data.reverse()
+      }
+      this.sortState = this.sortState === 'ascending' ? 'descending' : 'ascending'
+    }
```

Comment:
we should add a `sort` function to the `KTable` kongponent itself and then a user can just use that. It can be an exported function at the top of the KTable `<script>` tag similar to how we export appearances in `KButton`. Right now, none of our application specific logic uses anything special so a default function returning different comparators based on the data type even if it's just client side would be nice.